tests/php/test-class-amp-theme-support.php

```diff
@@ -2162,6 +2148,53 @@ public function test_prepare_response_doing_template_actions() {
 		}
 	}
 
+	/**
+	 * Test prepare_response for responses that throw an exception.
+	 *
+	 * @covers AMP_Theme_Support::prepare_response()
+	 */
+	public function test_prepare_response_throwing_exception() {
+		// Set up temporary capture of error log to test error log output.
+		$capture = tmpfile();
+		$backup  = ini_set(
+			'error_log',
+			stream_get_meta_data( $capture )['uri']
+		);
+
+		add_filter(
+			'amp_schemaorg_metadata',
+			static function () {
+				throw new RuntimeException( 'FAILURE', 42 );
+			}
+		);
+
+		if ( ! function_exists( 'newrelic_disable_autorum' ) ) {
+			/**
+			 * Define newrelic_disable_autorum to allow passing line.
+			 */
+			function newrelic_disable_autorum() {
+				return true;
+			}
+		}
+
+		wp();
+		$original_html = $this->get_original_html();
+		AMP_Theme_Support::start_output_buffering();
+		echo $original_html;
+		$output = ob_get_clean();
```

Comment:
This can be simplified:

```suggestion
		$output = AMP_Theme_Support::finish_output_buffering( $this->get_original_html() );
```